### PR TITLE
ENH: Flesh out ConcatenateLTA

### PR DESCRIPTION
**Pull request by effigies**
Changes proposed in this pull request
- Adds most options to `mri_concatenate_lta` (skipping `-rmsdiff` because it's a weird one and I don't need it)
- Update existing fields to match tool specs (see especially `in_lta2`)

### CODE REVIEW

**Comment by effigies** [comment]
A quick review of this one would be greatly appreciated.

File: nipype/interfaces/freesurfer/preprocess.py [hunk 1]
```diff
@@ -2397,23 +2422,44 @@ class ConcatenateLTAOutputSpec(TraitedSpec):
...
     _cmd = 'mri_concatenate_lta'
     input_spec = ConcatenateLTAInputSpec
     output_spec = ConcatenateLTAOutputSpec
 
+    def _format_arg(self, name, spec, value):
+        if name == 'out_type':
+            value = {'VOX2VOX': 0, 'RAS2RAS': 1}[value]
+        return super(ConcatenateLTA, self)._format_arg(name, spec, value)
+
     def _list_outputs(self):
```
Comment (satra) [inline nipype/interfaces/freesurfer/preprocess.py]: perhaps this can be removed?

Comment (effigies) [reply nipype/interfaces/freesurfer/preprocess.py]: I'm guessing something else would be needed to change? Removing `_list_outputs` gets me with an `<undefined>` `outputs.out_file`.

Comment (satra) [reply nipype/interfaces/freesurfer/preprocess.py]: `name_source` in outputspec

**Comment by satra** [comment]
@effigies - just merged oscar's terminal output PR, so make specs would need to be redone.

File: nipype/interfaces/freesurfer/preprocess.py (hunk 1 above)
Comment (effigies) [reply nipype/interfaces/freesurfer/preprocess.py]: I'm not able to figure out how to use `name_source`. Has no effect I can see in output spec, and trying to follow the examples in the [docs](https://nipype.readthedocs.io/en/latest/devel/interface_specs.html) results in failing to generate a string for `inputs.out_file` and the interface fails.

Comment (satra) [reply nipype/interfaces/freesurfer/preprocess.py]: nevermind - the logic i had in mind is only implemented if `name_source` is in the input trait. i think the next refactor needs to take care of filename generation

**Comment by effigies** [comment]
@satra Any other comments? Good to merge?

**Pull request merged**